ScanCommand/src/test/java/io/ballerina/scan/internal/ReporterImplTest.java

```diff
@@ -0,0 +1,93 @@
+/*
+ *  Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
+ *
+ *  WSO2 LLC. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied. See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package io.ballerina.scan.internal;
+
+import io.ballerina.compiler.syntax.tree.ModulePartNode;
+import io.ballerina.projects.Document;
+import io.ballerina.projects.Module;
+import io.ballerina.projects.Project;
+import io.ballerina.projects.directory.SingleFileProject;
+import io.ballerina.scan.Issue;
+import io.ballerina.scan.Rule;
+import io.ballerina.scan.Severity;
+import io.ballerina.scan.Source;
+import io.ballerina.tools.text.LineRange;
+import io.ballerina.tools.text.TextRange;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+import org.wso2.ballerinalang.compiler.diagnostic.BLangDiagnosticLocation;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ReporterImplTest {
+    @Test(description = "test creating a reporter and retrieving issues reported with a numeric rule identifier")
+    void testReporterWithNumericId() {
+        RuleImpl rule = new RuleImpl("B101", 101, "rule 101", Severity.BUG);
+        List<Rule> rules = new ArrayList<>();
+        rules.add(rule);
+        ReporterImpl reporter = new ReporterImpl(rules);
+        Path validBalProject = Paths.get("src", "test", "resources", "test-resources",
+                "valid-single-file-project", "main.bal");
+        Project project = SingleFileProject.load(validBalProject);
+        Module defaultModule = project.currentPackage().getDefaultModule();
+        Document document = defaultModule.document(defaultModule.documentIds().iterator().next());
+        ModulePartNode modulePartNode = (ModulePartNode) document.syntaxTree().rootNode();
+        LineRange lineRange = modulePartNode.location().lineRange();
+        TextRange textRange = modulePartNode.location().textRange();
+        BLangDiagnosticLocation location = new BLangDiagnosticLocation(lineRange.fileName(),
+                lineRange.startLine().line(), lineRange.endLine().line(),
+                lineRange.startLine().offset(), lineRange.endLine().offset(),
+                textRange.startOffset(), textRange.length());
+        reporter.reportIssue(document, location, 101);
+        List<Issue> issues = reporter.getIssues();
+        Issue issue = issues.get(0);
+        Assert.assertEquals(location, issue.location());
```

Comment:
`Assert.assertEqual(String actual, String expected)` in testng is following order. So we need to swap the variables
```suggestion
        Assert.assertEquals(issue.location(), location);
```

Reply:
Check other places as well